app/src/androidTest/java/com/example/android/justjava/Models/JustJavaModel.kt

```diff
@@ -0,0 +1,214 @@
+        verifyNameAdded()
+    }
+
+    // add a name greater than character limit set in activity_main.xml
+    private fun exceedCharLimit() {
+        nameEditText.perform(replaceText(longName))
+    }
+
+    // verify that only the correct characters were added(did not exceed char limit)
+    private fun verifyCharLimit() {
+        nameEditText.check(matches(withText(shortenedName)))
+    }
+
+    // attempt to add long name, verify only max chars added
+    fun exceedCharLimitAndVerify() {
+        exceedCharLimit()
+        verifyCharLimit()
+    }
+
+
+    private fun addNameToEditText(name: String) {
+        nameEditText.perform(replaceText(name))
+    }
+
+    // check the whipped cream checkbox
+    private fun clickWhippedCreamCheckbox() {
+        creamCheckBox.perform(click())
+    }
+
+    // check the chocolate checkbox
+    fun clickChocolateCheckbox() {
+        chocolateCheckBox.perform(click())
+
+    }
+
+    // verify cream checkbox has been checked
+    fun verifyCreamCheckbox() {
+        clickWhippedCreamCheckbox()
+        creamCheckBox.check(matches(isChecked()))
+    }
+
+    // verify both boxes can be checked
+    fun verifyBothChecked() {
+        clickChocolateCheckbox()
+        clickWhippedCreamCheckbox()
+        creamCheckBox.check(matches(isChecked()))
+        chocolateCheckBox.check(matches(isChecked()))
+
+    }
+
+    // verify BOTH boxes checked, unchecked, remain unchecked
+    fun verifyBothUnchecked() {
+        clickWhippedCreamCheckbox()
+        clickChocolateCheckbox()
+        clickWhippedCreamCheckbox()
+        clickChocolateCheckbox()
+        chocolateCheckBox.check(matches(isNotChecked()))
+        creamCheckBox.check(matches(isNotChecked()))
+    }
+
+    // click minus button
+    private fun clickMinusButton() {
+        minusButton.perform(click())
+    }
+
+    // verify that count cannot be less than 1, verify toast message
+    fun verifyMinCount() {
+        closeSoftKeyboard()
+        clickMinusButton()
+        clickMinusButton()
+        clickMinusButton()
+        clickMinusButton()
+        onView(withText(R.string.min_count_toast)).inRoot(ToastMatcher()).check(matches(withText(YOU_CANNOT_HAVE_LESS_THAN_1_COFFEE)))
+        quantityDisplay.check(matches(withText("1")))
+    }
+
+
+    // click plus button
+    private fun clickPlusButton() {
+        plusButton.perform(click())
+    }
+
+    // check that count has increased after clicking plus button
+    fun verifyIncreasedCount() {
+        clickPlusButton()
+        quantityDisplay.check(matches(withText("2")))
+    }
+
+    // check that count does not exceed 5, verify toast message
+    fun verifyMaxCount() {
+        closeSoftKeyboard()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        onView(withText(R.string.max_count_toast)).inRoot(ToastMatcher()).check(matches(withText(MAX_5_COFFEES_PER_ORDER)))
```

Comment:
verifyMinCount and verifyMaxCount... personally I don't care if you have a space before you verify the toast message but it's important to be consistent. So either here add a space before verifying the toast or in verifyMinCount remove the space before verifying the toast